hibernate-core/src/main/java/org/hibernate/loader/ast/internal/SingleIdLoadPlan.java

```diff
@@ -57,18 +55,14 @@ public SingleIdLoadPlan(
 		this.restrictivePart = restrictivePart;
 		this.lockOptions = lockOptions.makeCopy();
 		this.jdbcParameters = jdbcParameters;
+		this.sqlAst = sqlAst;
```

Comment:
Storing the whole SQL AST tree here means having to keep a complex structure around for every single-id load plan, which means for every mapped entity type. It would be best to avoid this, especially seems the `SubselectFetch#createRegistrationHandler` method apparently doesn't use it anymore apart from a specific case.

Reply:
I see what you mean here. Each SingleIdLoadPlan is stored in some sort of cache, it doesn't make much sense to keep around the queryast as well.